src/freenet/clients/fcp/ClientGet.java

```diff
@@ -183,7 +183,7 @@ public ClientGet(PersistentRequestClient globalClient, FreenetURI uri, boolean d
 			        targetFile.delete();
                     Logger.error(this, "Target file already exists but is zero length, deleting...");
 			    }
-			    if(targetFile.exists())
+			    else
 			        throw new IOException("Target filename exists already: "+targetFile);
```

Comment:
Your proposed change appears valid. Unfortunately this does not comply with the coding standards, in particular this lacks the necessary curly brackets.

Suggested alternative:
```java
    ...
    Logger.error(this, "Target file already exists but is zero length, deleting...");
} else {
    throw new IOException("Target filename exists already: " + targetFile);
}
...
```